drivers/pinctrl/pinctrl_mchp_xec.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ * Copyright (c) 2021 Linaro Limited
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_pinctrl
+
+#include <drivers/pinctrl.h>
+#include <gpio/gpio_mchp_xec.h>
+#include <soc.h>
+
+/*
+ * const struct device *dev = pointer to gpio_xxx_yyy device
+ * uint32_t pin = encoded port and pin number
+ * uint32_t cfg = property encodings: pulls, buffer type, slew, drive strength
+ * uint32_t altf = function number, 0 = GPIO, > 0 alternate function
+ */
+static int mchp_xec_pin_configure(const struct device *dev, uint32_t portpin,
+				  uint32_t cfg, uint32_t altf)
+{
+	uint32_t pin = MCHP_XEC_PINMUX_PIN(portpin);
+
+	if (!dev || !device_is_ready(dev)) {
+		return -ENODEV;
+	}
```

Comment:
related to previous comment: this checks for device readyness every time state is configured, doesn't make much sense.

Reply:
will remove device_is_ready() check

Reply:
updated